### ISSUE TITLE
lodash-es isEqual very broken in Bun

### ISSUE THREAD

**Issue by dtinth** [oven-sh/bun#2439]
### What version of Bun is running?

0.5.8

### What platform is your computer?

Linux 5.15.63 x86_64 unknown

### What steps can reproduce the bug?

https://codesandbox.io/p/sandbox/infallible-elbakyan-kuz59j?file=%2FREADME.md

Install `lodash-es`

```sh
bun add lodash-es
```

Create `test.mjs`

```js
import { isEqual } from "lodash-es";

console.log(isEqual({ a: 1 }, { a: 2 }));
```

Run `test.mjs` with bun:

```sh
bun test.mjs
```

### What is the expected behavior?

Should output `false` because object is different.

### What do you see instead?

It outputs `true`. None of the JavaScript runtimes I tried exhibit this weird behavior.

### Additional information

Here’s an example screenshot with Node vs Bun. Also reproducable on my M1 MacBook.

<img width="347" alt="image" src="https://user-images.githubusercontent.com/193136/226416958-3929744d-d660-4dc8-9d96-23b69baa1765.png">

**Comment by Jarred-Sumner** [comment]
This is a CommonJS <> ESM interop bug. `isBuffer` is returning the CommonJS transformed version which is unexpected and returning a truthy value for every input

**Comment by SaltyAom** [comment]
Seem to be fixed on 0.6.0 (canary).

Darwin Salt.local 22.3.0 Darwin Kernel Version 22.3.0: Mon Jan 30 20:38:37 PST 2023; root:xnu-8792.81.3~2/RELEASE_ARM64_T6000 arm64

<img width="816" alt="Screenshot 2566-05-11 at 22 14 09" src="https://github.com/oven-sh/bun/assets/35027979/ebd374ea-c7cd-4982-8f75-ba4bc69c25bf">

**Comment by paperclover** [comment]
this seems to be fixed in 0.6 but there are issues when bundling lodash-es, tracking that in #3206
